api/v1alpha1/azurevalidator_types.go

```diff
@@ -55,6 +58,31 @@ type RBACRule struct {
 	PrincipalID string `json:"principalId" yaml:"principalId"`
 }
 
+// Conveys that one or more images in a community gallery exist.
+type CommunityGalleryImageRule struct {
+	// Unique identifier for the rule in the validator. Used to ensure conditions do not overwrite
+	// each other.
+	Name string `json:"name" yaml:"name"`
+	// Gallery is the community gallery.
+	Gallery CommunityGallery `json:"gallery" yaml:"gallery"`
+	// Images are the names of the images to check for in the gallery.
+	//+kubebuilder:validation:MinItems=1
+	//+kubebuilder:validation:MaxItems=20
```

Comment:
What's the rationale for capping this at 20?

Reply:
I knew I wanted a "max items" validation and I saw one that was 1-20 on pre-existing part of the spec, so I copied it over. Now that I think about it though, this new rule is less complex than the old one, and we can use a high limit for number of images. I'll change this.